modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/CommitWriteInvokeClosure.java

```diff
@@ -61,11 +68,23 @@ public void call(@Nullable VersionChain oldRow) throws IgniteInternalCheckedExce
             return;
         }
 
-        updateTimestampLink = oldRow.headLink();
-
         operationType = OperationType.PUT;
 
-        newRow = VersionChain.createCommitted(oldRow.rowId(), oldRow.headLink(), oldRow.nextLink());
+        RowVersion current = storage.readRowVersion(oldRow.headLink(), ALWAYS_LOAD_VALUE, false);
+        RowVersion next = oldRow.hasNextLink() ? storage.readRowVersion(oldRow.nextLink(), ALWAYS_LOAD_VALUE, false) : null;
+
+        // If the previous and current version are tombstones, then delete the current version.
+        if (next != null && current.isTombstone() && next.isTombstone()) {
```

Comment:
So, I guess you always load value to check if it's a tombstone or not. If this is correct, then it's like writing "size() == 0" instead of "isEmpty()", makes no sense. If this is not correct, then I apologize

Reply:
I didn’t quite understand you for `next`, I don’t load data (`ByteBuffer`), but only read field `RowVersion#valueSize` and find out from it whether it is a tombstone or not.

Reply:
Fix it